### PR TITLE
PLU-261: [onboarding-demo-experiment-4]: add demo video and the tooltip inside flow editor

### PR DESCRIPTION
**Pull request by m0nggh**
## Problem
New users are confused with how to get started on Plumber, so GGWP v1 is an onboarding demo flow we are experimenting with to provide more help for users to onboard with Plumber.

## Details
Frontend
- The demo video appears on every pipe, pulsating effect only appears the first time based on `localStorage`

Backend
- Add the demo video attributes to every app, only formsg and postman will have the actual links now

## Tests
- [x] All pipes should see the demo video tooltip and video should appear upon clicking on it
- [x] Only formsg and postman steps will have the video tooltip

### CODE REVIEW

**Comment by m0nggh** [comment]
* **#622** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/622?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#621** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/621?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#620** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/620?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#619** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/619?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#618** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/618?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#617** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/617?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @m0nggh and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/src/components/FlowStepHeader/index.tsx [hunk 1]
```diff
@@ -57,122 +76,181 @@ export default function FlowStepHeader(
...
+  const [hasSeenDemo, setHasSeenDemo] = useState<string | null>(
+    localStorage.getItem(LOCAL_STORAGE_HAS_SEEN_DEMO_KEY),
+  )
+
+  const handleDemoClick = useCallback(
+    (event: MouseEvent) => {
+      event.stopPropagation()
+      onModalOpen()
+      localStorage.setItem(LOCAL_STORAGE_HAS_SEEN_DEMO_KEY, 'true')
+      setHasSeenDemo('true')
```
Comment (pregnantboy) [inline packages/frontend/src/components/FlowStepHeader/index.tsx]: setHasSeenDemo(true)

File: packages/frontend/src/components/FlowStepHeader/index.tsx [hunk 2]
```diff
@@ -57,122 +76,181 @@ export default function FlowStepHeader(
...
+  const {
+    isOpen: isModalOpen,
+    onOpen: onModalOpen,
+    onClose: onModalClose,
+  } = useDisclosure()
+  const hasDemoVideo = !!demoVideoUrl && !!demoVideoTitle
+
+  // check whether user has opened the demo tooltip previously
+  const [hasSeenDemo, setHasSeenDemo] = useState<string | null>(
+    localStorage.getItem(LOCAL_STORAGE_HAS_SEEN_DEMO_KEY),
```
Comment (pregnantboy) [inline packages/frontend/src/components/FlowStepHeader/index.tsx]:
```suggestion
    localStorage.getItem(LOCAL_STORAGE_HAS_SEEN_DEMO_KEY) === 'true',
```

File: packages/frontend/src/components/FlowStepHeader/index.tsx [hunk 3]
```diff
@@ -57,122 +76,181 @@ export default function FlowStepHeader(
...
+  // for loading demo modal
+  const {
+    isOpen: isModalOpen,
+    onOpen: onModalOpen,
+    onClose: onModalClose,
+  } = useDisclosure()
+  const hasDemoVideo = !!demoVideoUrl && !!demoVideoTitle
+
+  // check whether user has opened the demo tooltip previously
+  const [hasSeenDemo, setHasSeenDemo] = useState<string | null>(
```
Comment (pregnantboy) [inline packages/frontend/src/components/FlowStepHeader/index.tsx]:
```suggestion
  const [hasSeenDemo, setHasSeenDemo] = useState<boolean>(
```

File: packages/frontend/src/components/FlowStepHeader/index.tsx [hunk 4]
```diff
@@ -57,122 +76,181 @@ export default function FlowStepHeader(
...
+                  openDelay={300}
+                >
+                  <Box boxSize={5}>
+                    <Icon
+                      as={BiHelpCircle}
+                      boxSize="inherit"
+                      sx={{
+                        borderRadius: '50%',
+                        animation:
+                          hasSeenDemo !== 'true'
```
Comment (pregnantboy) [inline packages/frontend/src/components/FlowStepHeader/index.tsx]:
```suggestion
                          hasSeenDemo
```

File: packages/frontend/src/components/FlowStepHeader/index.tsx [hunk 5]
```diff
@@ -31,8 +44,12 @@ interface FlowStepHeaderProps {
   onClose: () => void
   collapsed: boolean
   children: ReactNode
+  demoVideoUrl?: string
+  demoVideoTitle?: string
 }
 
+const LOCAL_STORAGE_HAS_SEEN_DEMO_KEY = 'has-seen-demo-feature'
```
Comment (pregnantboy) [inline packages/frontend/src/components/FlowStepHeader/index.tsx]:
```suggestion
const LOCAL_STORAGE_DEMO_TOOLTIP_KEY = 'demo-tooltip-clicked'
```

File: packages/frontend/src/components/FlowStepHeader/index.tsx [hunk 6]
```diff
@@ -57,122 +76,181 @@ export default function FlowStepHeader(
...
+              {hintAboveCaption}
+            </Text>
+
+            <Flex alignItems="center" gap={4}>
+              <Text textStyle="subhead-1" color="base.content.default">
+                {caption}
+              </Text>
+              {hasDemoVideo && (
+                <Tooltip
+                  label="Demo video"
```
Comment (pregnantboy) [inline packages/frontend/src/components/FlowStepHeader/index.tsx]:
```suggestion
                  label="Learn how to set this up"
```

**Review by pregnantboy — COMMENTED** [review]
[uploading image.png...]

**Review by pregnantboy — APPROVED** [review]
approve with nits

File: packages/frontend/src/components/FlowStepHeader/index.tsx (hunk 4 above)
Comment (m0nggh) [reply packages/frontend/src/components/FlowStepHeader/index.tsx]: ok i flip this logic

**Pull request merged**